src/main/java/org/gwtproject/tools/cldr/CurrencyListProcessor.java

```diff
@@ -0,0 +1,366 @@
+/*
+ * Copyright 2010 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.gwtproject.tools.cldr;
+
+import com.squareup.javapoet.*;
+import org.gwtproject.i18n.shared.GwtLocale;
+import org.gwtproject.i18n.shared.cldr.CurrencyData;
+import org.gwtproject.i18n.shared.cldr.CurrencyList;
+import org.unicode.cldr.util.XPathParts;
+
+import javax.lang.model.element.Modifier;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.util.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+/**
+ * Loads data needed to produce DateTimeFormatInfo implementations.
+ */
+public class CurrencyListProcessor extends Processor {
+
+    private static final Logger LOGGER = Logger.getLogger(CurrencyListProcessor.class.getName());
+
+    private Map<String, Integer> currencyFractions = new HashMap<String, Integer>();
+    private int defaultCurrencyFraction;
+    private Map<String, Integer> rounding = new HashMap<String, Integer>();
+
+    private Set<String> stillInUse = new HashSet<String>();
+    private Map<GwtLocale, List<LocaleData.Currency>> localesCurrencies;
+    private Map<String, String> languageMostPopulatedRegion;
+    private Map<String, String> regionsCurrencies;
+
+    public CurrencyListProcessor(File outputDir, InputFactory cldrFactory, LocaleData localeData) {
+        super(outputDir, cldrFactory, localeData);
+    }
+
+    @Override
+    protected void cleanupData() {
+//        localeData.removeDuplicates("currency");
```

Comment:
?

Reply:
I thought i had to do this for this processor just as some other processors did, but it didnt work as expected, i should have removed the comment.